picker/picker_test.go

```diff
@@ -262,6 +262,65 @@ func TestRemotesDownweight(t *testing.T) {
 	}
 }
 
+// TestRemotesPractical ensures that under a single poor observation, such as
+// an error, the likelihood of selecting the node dramatically decreases.
+func TestRemotesPractical(t *testing.T) {
+	peers := []api.Peer{{Addr: "one"}, {Addr: "two"}, {Addr: "three"}}
+	remotes := NewRemotes(peers...)
+	seen := map[api.Peer]int{}
+	selections := 1000
+	tolerance := 0.04
+
+	// set a baseline, where selections should be even
+	for i := 0; i < selections; i++ {
+		peer, err := remotes.Select()
+		if err != nil {
+			t.Fatalf("error selecting peer: %v", err)
+		}
+
+		seen[peer]++
+	}
+
+	expected, delta := selections/len(peers), int(tolerance*float64(selections))
+	low, high := expected-delta, expected+delta
+	for peer, count := range seen {
+		if !(count >= low && count <= high) {
+			t.Fatalf("weighted selection not balanced: %v selected %v/%v, expected range %v, %v", peer, count, selections, low, high)
+		}
+	}
+
+	// one bad observation should mark the node as bad
+	remotes.Observe(peers[0], -DefaultObservationWeight)
```

Comment:
You can either have balanced downweighting have failures immediately reduce selection probability or have less negative than positive to require several observations before it is fully downweighted.

In this case, initial condition is `DefaultObservationWeight` and we downweight with `-DefaultObservationWeight`, expecting it to cross zero.